int32 x_control
int32 z_control
int32 angle_control
int32 grabe_control
int32 close_machine
